components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -451,9 +456,9 @@ public Blob requireBlob(final BlobRef blobRef) {
     return blob;
   }
 
-  private void deleteBlob(final BlobRef blobRef, boolean checkWritePolicy) {
+  private void deleteBlob(final Asset asset, final BlobRef blobRef, boolean checkWritePolicy) {
```

Comment:
Could just pass in the effective write policy here which avoids the need to pass in the asset and select the policy again. ie. `deleteBlob(final BlobRef blobRef, WritePolicy writePolicy) {` - clients would then select the policy or pass in `null` if the policy shouldn't be checked.